src/vs/workbench/contrib/terminal/browser/xterm/textAreaSyncAddon.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+import { ITerminalCapabilityStore, TerminalCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+import type { Terminal, ITerminalAddon } from 'xterm';
+
+export interface ITextAreaData {
+	content: string;
+	cursorX: number;
+}
+
+export class TextAreaSyncAddon extends Disposable implements ITerminalAddon {
+	private _terminal: Terminal | undefined;
+	private readonly _onDidRequestUpdateTextArea = this._register(new Emitter<ITextAreaData>());
+	readonly onDidRequestUpdateTextArea = this._onDidRequestUpdateTextArea.event;
+	activate(terminal: Terminal): void {
+		this._terminal = terminal;
+		this._register(this._terminal.onCursorMove(() => this._refreshTextArea()));
```

Comment:
You should listen to accessibility mode changed and only have this listener installed when it's on, otherwise we have an event handler firing a lot without getting handled. `MutableDisposable` will be handy for keeping the optional onCursorMove listener.